OB-996: add document about new asset code (#83)
akeneo/onboarder-helpcenter: content/md/retailer-onboarder/define-suppliers-scope/define-assets-requirements.md

If the attribute used as the main media of the asset family linked to this asset collection is a media link attribute, the collection won't be synchronized on the supplier's end and they won't be able to send the assets.
:::

##  New asset code uniqueness

> Before, asset code was generating using only UUID, now there is a new asset code generation

Reviewer: ```suggestion
Before, asset code was generated using only UUID, now there is a new asset code generation.
```

Reviewer: Can you be more specific? Is it the generation you explain below?

As many suppliers can be working on the same asset collection at the same time, they could set the same asset code twice. The Onboarder puts a constraint on suppliers to guarantee the uniqueness of asset codes among all suppliers.
* A supplier can't create an asset code

> * The asset code is automatically generated using a concatenation of :

Reviewer: ```suggestion
* The asset code is automatically generated using a concatenation of:
```

  - supplier code
  - filename

>   - uuid

Reviewer: ```suggestion
  - UUID
```

> For example, the supplier 'pac' upload new asset called brown-chair.jpg, the asset code will be something like :

Reviewer: ```suggestion
For example, the supplier 'pac' uploads a new asset called brown-chair.jpg, the asset code will be something like:
```

 - pac_brown_chair_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg

> You can easily create a rule on it, you can guess from the beginning of the asset code, which supplier added it

Reviewer: Not clear for me :/

> Regex could be : ```/^pac_/``` meaning the asset code must start by this string

Reviewer: ```suggestion
Regex could be : ```/^pac_/```, meaning the asset code must start with this string.
```

> Regex could be : ```/^pac_/```, meaning the asset code must start with this string

Reviewer 2: Thank you for this PR :) 

The multiline commit suggestion didn't work when there is line deletion, so here is the changes below and my comments here:
- The example isn't enough clear enough for a user to setup a product link rule
- Actually, the asset code format isn't present

```
## Asset code format and uniqueness

As many suppliers can be working on the same asset collection at the same time, they could have the same asset code twice. The Onboarder puts a constraint on suppliers to guarantee the uniqueness of asset codes among all suppliers.
* A supplier can't create an asset code
* The asset code is automatically generated by replacing hyphens with underscores and by using a concatenation of:
  - supplier code
  - filename
  - UUID

Here is the asset code format: `<supplier_code>_<filename>_<uuid>`

For example, the supplier `pac` upload a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`

You can easily create product link rules to automatically link assets to products/product models based on assets filenames (More info on [https://help.akeneo.com/pim/serenity/articles/assets-product-link-rules.html](product link rules)).
On our example, regex could be: `(?<supplier_code>.*?)_(?<product_ref>.*?)_(?<attribute_ref>.*?)_(.*).jpg`

::: warning
Previous uploaded assets will still have the old asset code format, you need to reupload them to have the new format.
::::
```

> For example, the supplier `pac` upload a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`

Reviewer 2: ```suggestion
For example, the supplier `pac` uploads a new asset called `chair-packshot.jpg`, the asset code will be: `pac_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```

> For example, the supplier `supplier1` upload a new asset called `chair-packshot.jpg`, the asset code will be: `supplier1_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`

Reviewer 2: ```suggestion
For example, the supplier `supplier1` uploads a new asset called `chair-packshot.jpg`, the asset code will be: `supplier1_chair_packshot_8d704905_fbf9_42b9_94af_eb399dbaaaf2.jpg`
```
